src/ChunkHTTPServer.cpp

```diff
@@ -0,0 +1,225 @@
+
+#include "Globals.h"  // NOTE: MSVC stupidness requires this to be the same across all modules
+
+#include "ChunkHTTPServer.h"
+
+#include "HTTP/HTTPMessage.h"
+#include "HTTP/HTTPServerConnection.h"
+
+#include "WorldStorage/SchematicFileSerializer.h"
+#include "WorldStorage/NBTChunkSerializer.h"
+#include "WorldStorage/FastNBT.h"
+#include "StringCompression.h"
+
+#include <regex>
+
+
+
+
+
+static const char DEFAULT_CHUNCKHTTPSERVER_PORTS[] = "5555";
+
+
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// ChunkHTTPServer:
+
+ChunkHTTPServer::ChunkHTTPServer(void) :
+	m_IsInitialized(false),
+	m_IsRunning(false)
+{
+}
+
+
+
+
+
+ChunkHTTPServer::~ChunkHTTPServer()
+{
+	ASSERT(!m_IsRunning);  // Was the HTTP server stopped properly?
+}
+
+
+
+
+
+bool ChunkHTTPServer::Init(void)
+{
+	LOGD("Initialising Chunck HTTP Server...");
+
+	if (!m_HTTPServer.Initialize())
+	{
+		return false;
+	}
+
+	m_IsInitialized = true;
+
+	return true;
+}
+
+
+
+
+
+bool ChunkHTTPServer::Start(void)
+{
+	if (!m_IsInitialized)
+	{
+		// Not initialized
+		return false;
+	}
+
+	LOGD("Starting Chunck HTTP Server...");
+
+	AStringVector Ports =
+	{
+		DEFAULT_CHUNCKHTTPSERVER_PORTS
+	};
+
+	m_IsRunning = m_HTTPServer.Start(*this, Ports);
+	return m_IsRunning;
+}
+
+
+
+
+
+void ChunkHTTPServer::Stop(void)
+{
+	if (!m_IsRunning)
+	{
+		return;
+	}
+
+	LOGD("Stopping Chunck HTTP Server...");
+	m_HTTPServer.Stop();
+	m_IsRunning = false;
+}
+
+
+
+
+
+class ChunkHTTPServerUserData:
+	public cHTTPIncomingRequest::cUserData
+{
+	public:
+	AString m_Data;
+};
+
+
+
+
+void ChunkHTTPServer::OnRequestBegun(cHTTPServerConnection & a_Connection, cHTTPIncomingRequest & a_Request)
+{
+	UNUSED(a_Connection);
+
+	LOGD("OnRequestBegun");
+
+	a_Request.SetUserData(std::make_shared<ChunkHTTPServerUserData>());
+}
+
+
+
+
+
+void ChunkHTTPServer::OnRequestBody(cHTTPServerConnection & a_Connection, cHTTPIncomingRequest & a_Request, const char * a_Data, size_t a_Size)
+{
+	UNUSED(a_Connection);
+
+	LOGD("OnRequestBody (%lld)", (long long)a_Size);
```

Comment:
Please try not to use builtin integer types aside from int/unsigned for general use and size_t for sizes. Instead use the defined IntXX or UIntXX types.

Reply:
While I agree with you concerning the use of (u)intXX, concerning the printf format : 
"%lld" = long long int decimal
Even if I know that long long always correspond to int64_t.
I just can't make a mistake while writing this way.

Anyway this log is not so useful... Maybe it's better to remove it.